### PR TITLE
M3-2631: Fix TSLint warnings

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

** **Please read the notes at the bottom of this description.** **

Fixes the following tslint warnings:

```
WARNING: src/components/PaginationControls/PageNumbers.tsx:69:1 - asterisks in jsdoc must be aligned
WARNING: src/components/PaginationControls/PageNumbers.tsx:70:1 - asterisks in jsdoc must be aligned
WARNING: src/components/PaginationControls/PageNumbers.tsx:71:1 - asterisks in jsdoc must be aligned
WARNING: src/components/StatusIndicator/index.ts:2:18 - An interface declaring no members is equivalent to its supertype.
WARNING: src/features/Billing/PdfGenerator/PdfGenerator.ts:150:11 - Class name must be in pascal case
WARNING: src/features/Billing/PdfGenerator/PdfGenerator.ts:263:5 - Calls to 'console.error' are not allowed.
WARNING: src/features/Billing/PdfGenerator/PdfGenerator.ts:358:5 - Calls to 'console.error' are not allowed.
WARNING: src/features/Domains/DomainDetail.tsx:159:14 - Calls to 'console.error' are not allowed.
WARNING: src/features/Domains/DomainDetail.tsx:176:14 - Calls to 'console.error' are not allowed.
WARNING: src/features/NodeBalancers/NodeBalancerDetail/NodeBalancerDetail.tsx:225:22 - Shadowed name: 'p'
WARNING: src/features/Profile/APITokens/APITokenTable.tsx:466:1 - asterisks in jsdoc must be aligned
WARNING: src/features/Profile/APITokens/APITokenTable.tsx:467:1 - asterisks in jsdoc must be aligned
WARNING: src/features/Profile/APITokens/APITokenTable.tsx:468:1 - asterisks in jsdoc must be aligned
WARNING: src/features/Profile/APITokens/APITokenTable.tsx:469:1 - asterisks in jsdoc must be aligned
WARNING: src/features/Profile/APITokens/APITokenTable.tsx:470:1 - asterisks in jsdoc must be aligned
WARNING: src/features/Profile/APITokens/APITokenTable.tsx:471:1 - asterisks in jsdoc must be aligned
WARNING: src/features/Profile/APITokens/APITokenTable.tsx:472:1 - asterisks in jsdoc must be aligned
WARNING: src/features/Profile/AuthenticationSettings/ResetPassword.tsx:14:11 - An empty interface is equivalent to `{}`.
WARNING: src/features/TopMenu/UserEventsMenu/UserEventsList.tsx:20:7 - Calls to 'console.log' are not allowed.
WARNING: src/features/linodes/LinodesDetail/LinodeRebuild/RebuildFromStackScript.tsx:203:28 - object access via string literals is disallowed
WARNING: src/features/linodes/LinodesDetail/initLinode.ts:21:25 - Shadowed name: 'getLinode'
WARNING: src/services/linodes/getLinodeInfo.ts:7:6 - Use an interface instead of a type literal.
WARNING: src/store/events/event.helpers.test.ts:284:46 - block is empty
WARNING: src/store/events/event.helpers.ts:16:1 - asterisks in jsdoc must be aligned
WARNING: src/store/events/event.helpers.ts:16:1 - jsdoc is not formatted correctly on this line
WARNING: src/store/events/event.reducer.test.ts:79:45 - block is empty
WARNING: src/store/events/event.reducer.test.ts:83:33 - block is empty
WARNING: src/store/events/event.reducer.test.ts:84:35 - block is empty
WARNING: src/utilities/isCreditCardExpired.ts:23:19 - jsdoc is not formatted correctly on this line
```

## Note to Reviewers

**To test,** Run `yarn lint`. There should be no errors.

In particular, please check the changes in `DomainDetail.tsx`. All we were doing in `catch` blocks was `console.error`. It was really old code, so I changed it to actually set the error to state.

Also, JSDoc-style comments within JSX don't play well with Prettier and TSLint, so we should avoid these in the future.

### CODE REVIEW

File: src/features/Domains/DomainDetail.tsx [hunk 1]
```diff
@@ -151,12 +151,12 @@ class DomainDetail extends React.Component<CombinedProps, State> {
     if (!domainId) {
       return;
     }
-
     getAllWithArguments<Linode.DomainRecord>(getDomainRecords)([+domainId])
       .then(({ data }) => {
         this.setState({ records: data });
+        throw new Error();
```
Comment (martinmckenna) [inline src/features/Domains/DomainDetail.tsx]: oh wait this needs to be removed

File: src/features/Domains/DomainDetail.tsx [hunk 2]
```diff
@@ -172,8 +172,9 @@ class DomainDetail extends React.Component<CombinedProps, State> {
     getDomain(+domainId)
       .then((data: Linode.Domain) => {
         this.setState({ domain: data });
+        throw new Error();
```
Comment (martinmckenna) [inline src/features/Domains/DomainDetail.tsx]: same with this

File: src/features/Domains/DomainDetail.tsx [hunk 3]
```diff
@@ -151,12 +151,12 @@ class DomainDetail extends React.Component<CombinedProps, State> {
...
       return;
     }
-
     getAllWithArguments<Linode.DomainRecord>(getDomainRecords)([+domainId])
       .then(({ data }) => {
         this.setState({ records: data });
+        throw new Error();
       })
-      .catch(console.error);
+      .catch(e => this.setState({ error: e }));
```
Comment (martinmckenna) [inline src/features/Domains/DomainDetail.tsx]: Also this is debatable - Do we want to bomb the entire page if the user created a domain record and then the request to get the refreshed list of records fails?

File: src/features/Domains/DomainDetail.tsx (hunk 1 above)
Comment (johnwcallahan) [reply src/features/Domains/DomainDetail.tsx]: 🤦‍♂️ good catch @martinmckenna ; addressed.

File: src/features/Domains/DomainDetail.tsx (hunk 3 above)
Comment (johnwcallahan) [reply src/features/Domains/DomainDetail.tsx]: Yeah, good point. I changed it back and just `tslint:disabled` the line.

**Pull request merged**